Distraction Shield/init.js

```diff
@@ -1,74 +1,93 @@
-
-/* --------------- ---- Session initializer ---- ---------------*/
-
-//First receive the blacklist and settings from the sync storage,
-//then create a onBeforeRequest listener using this list and the settings.
-initSession = function () {
-    // Settings need to be loaded before the listener is replaced. The replaceListener
-    // requires the blocked sites to be loaded, so these weird callbacks are required.
-    storage.getSettings(function(settings) {
-        settings.reInitTimer();
-        setLocalSettings(settings);
-        retrieveBlockedSites(replaceListener);
-    });
-};
+import {setLocalSettings, retrieveBlockedSites, replaceListener} from './background';
+import * as storage from './modules/storage';
+import BlockedSiteList from './classes/BlockedSiteList';
+import UserSettings from './classes/UserSettings';
+import Tracker from './modules/statistics/tracker';
+import * as constants from'./constants';
 
 /* --------------- ---- Run upon installation ---- ---------------*/
 
-chrome.runtime.onInstalled.addListener(function() {
-    storage.getAll( function(output) {
+/**
+ * function to be fired only when the extension is installed or updated. It initiates all the data and the storage.
+ * Furthermore it shows the intro tour and initializes the extension upon completeion.
+ */
+chrome.runtime.onInstalled.addListener((details) => {
+    storage.getAllUnParsed((output) => {
         initBlacklist(output.tds_blacklist);
         initInterceptCounter(output.tds_interceptCounter);
         initInterceptDateList(output.tds_interceptDateList);
         initExerciseTime(output.tds_exerciseTime);
         initSettings(output.tds_settings);
-        runIntroTour();
+        if (details.reason == 'install') {
+            dataCollectionMsg ();
+            runIntroTour();
+        }
     });
 });
 
-initBlacklist = function(list) {
+function initBlacklist(list) {
```

Comment:
we could do that just as well, but that results in a sloppy function for the chrome.runtime.onInstalled. Which will probably result in "too many levels of abstraction". Since it will be:
 if(! list) {
   initBlacklist;
} if (! settings) {
initSettings;
} and that about 5 times in a row